activejob/test/cases/railtie_test.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require "helper"
+require "active_support/testing/isolation"
+
+class RailtieTest < ActiveSupport::TestCase
+  include ActiveSupport::Testing::Isolation
+
+  setup do
+    require "rails"
+
+    @rails_logger = Logger.new(nil)
+
+    @app ||= Object.const_set(
+      "AJRailtieTestApp#{SecureRandom.alphanumeric(6)}",
+      Class.new(::Rails::Application)
+    ) do |app|
```

Comment:
I think, we can just use 
```ruby
@app ||= Class.new(::Rails::Application) do
```
here

Reply:
I tried this initially, but it fails with the following error:
```
Error:
RailtieTest#test_active_job.logger_initializer_does_not_overwrite_the_supplied_logger:
ArgumentError: Invalid app name. App names must be valid URI hostnames: alphanumeric and hyphen characters only.
    .../.asdf/installs/ruby/3.2.1/lib/ruby/gems/3.2.0/gems/globalid-1.2.1/lib/global_id/uri/gid.rb:51:in `rescue in validate_app'
    .../.asdf/installs/ruby/3.2.1/lib/ruby/gems/3.2.0/gems/globalid-1.2.1/lib/global_id/uri/gid.rb:48:in `validate_app'
    .../.asdf/installs/ruby/3.2.1/lib/ruby/gems/3.2.0/gems/globalid-1.2.1/lib/global_id/global_id.rb:33:in `app='
    .../.asdf/installs/ruby/3.2.1/lib/ruby/gems/3.2.0/gems/globalid-1.2.1/lib/global_id/railtie.rb:21:in `block in <class:Railtie>'
    .../forks/rails-fork/railties/lib/rails/initializable.rb:32:in `instance_exec'
    .../forks/rails-fork/railties/lib/rails/initializable.rb:32:in `run'
    .../forks/rails-fork/railties/lib/rails/initializable.rb:61:in `block in run_initializers'
    .../.asdf/installs/ruby/3.2.1/lib/ruby/3.2.0/tsort.rb:228:in `block in tsort_each'
    .../.asdf/installs/ruby/3.2.1/lib/ruby/3.2.0/tsort.rb:350:in `block (2 levels) in each_strongly_connected_component'
    .../.asdf/installs/ruby/3.2.1/lib/ruby/3.2.0/tsort.rb:431:in `each_strongly_connected_component_from'
    .../.asdf/installs/ruby/3.2.1/lib/ruby/3.2.0/tsort.rb:349:in `block in each_strongly_connected_component'
    .../.asdf/installs/ruby/3.2.1/lib/ruby/3.2.0/tsort.rb:347:in `each'
    .../.asdf/installs/ruby/3.2.1/lib/ruby/3.2.0/tsort.rb:347:in `call'
    .../.asdf/installs/ruby/3.2.1/lib/ruby/3.2.0/tsort.rb:347:in `each_strongly_connected_component'
    .../.asdf/installs/ruby/3.2.1/lib/ruby/3.2.0/tsort.rb:226:in `tsort_each'
    .../.asdf/installs/ruby/3.2.1/lib/ruby/3.2.0/tsort.rb:205:in `tsort_each'
    .../forks/rails-fork/railties/lib/rails/initializable.rb:60:in `run_initializers'
    .../forks/rails-fork/railties/lib/rails/application.rb:423:in `initialize!'
    .../forks/rails-fork/railties/lib/rails/railtie.rb:226:in `public_send'
    .../forks/rails-fork/railties/lib/rails/railtie.rb:226:in `method_missing'
    .../forks/rails-fork/activejob/test/cases/railtie_test.rb:31:in `block in <class:RailtieTest>'
```

Seems like GlobalID is [being used in the AJ test helper](https://github.com/rails/rails/blob/1f0262aa2b768b14de5e6ef29d0e547628f570ee/activejob/test/helper.rb#L9), which has a railtie [that extract Rails app's name](https://github.com/rails/globalid/blob/main/lib/global_id/railtie.rb#L20C1-L20C1).